XDCx/XDCx.go

```diff
@@ -579,17 +579,16 @@ func (XDCX *XDCX) GetEmptyTradingState() (*tradingstate.TradingStateDB, error) {
 func (XDCx *XDCX) GetStateCache() tradingstate.Database {
 	return XDCx.StateCache
 }
+
 func (XDCx *XDCX) HasTradingState(block *types.Block, author common.Address) bool {
 	root, err := XDCx.GetTradingStateRoot(block, author)
 	if err != nil {
 		return false
 	}
 	_, err = XDCx.StateCache.OpenTrie(root)
-	if err != nil {
-		return false
-	}
-	return true
+	return err == nil
```

Comment:
I would like the original way because it make the error handling consistent.
New way looks neat but I will it's more straight forward to read from original way.
```
	if err != nil {
		return false
	}
	return true
```
```
	return err == nil

```

Reply:
The staticcheck suggest:

```
XDCx/XDCx.go:588:2: should use 'return err == nil' instead of 'if err != nil { return false }; return true' (S1008)
```

![1700632551777](https://github.com/XinFinOrg/XDPoSChain/assets/7695325/316ffdeb-a23c-4dc4-a605-da177a68238f)

Maybe `return err == nil` can improve performance a little bit.